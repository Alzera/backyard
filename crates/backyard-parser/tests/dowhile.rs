use backyard_parser::parse_eval;

#[test]
fn basic() {
  let asts = parse_eval("do {\n} while (false);").unwrap();
  insta::assert_yaml_snapshot!(asts);
}

#[test]
fn separated_by_comment() {
  let asts = parse_eval("do {\n}\n// this comment\nwhile (false);").unwrap();
  insta::assert_yaml_snapshot!(asts);
}
