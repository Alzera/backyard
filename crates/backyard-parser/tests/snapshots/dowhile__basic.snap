---
source: crates/backyard-parser/tests/dowhile.rs
expression: asts
snapshot_kind: text
---
node_type: program
children:
  - node_type: do_while
    condition:
      node_type: do_while_condition
      condition:
        node_type: boolean
        is_true: false
        loc:
          start:
            line: 2
            column: 9
            offset: 14
          end:
            line: 2
            column: 9
            offset: 14
        leadings: ~
        trailings: ~
      loc:
        start:
          line: 2
          column: 2
          offset: 7
        end:
          line: 2
          column: 9
          offset: 14
      leadings: ~
      trailings: ~
    body:
      node_type: block
      statements: []
      loc:
        start:
          line: 1
          column: 3
          offset: 3
        end:
          line: 2
          column: 0
          offset: 5
      leadings: ~
      trailings: ~
    loc:
      start:
        line: 1
        column: 0
        offset: 0
      end:
        line: 2
        column: 14
        offset: 19
    leadings: ~
    trailings: ~
loc:
  start:
    line: 1
    column: 0
    offset: 0
  end:
    line: 2
    column: 15
    offset: 20
leadings: ~
trailings: ~
